### ISSUE TITLE
shown number of records mismatched with uploaded file

### ISSUE THREAD

**Issue by hrnciar** [ckan/ckan#5352]
### CKAN Version if known (or site URL) 
2.8.1
http://sc02.fi.muni.cz/en/dataset/teplota/resource/e5f5ef84-deb3-464d-a130-8a03c3a8a6cf

### Please describe the expected behaviour
Uploaded file has 34945 lines, I would expect that number of records is 34945.

### Please describe the actual behaviour
The number of records is 26204 (this numbers differs when I run my code multiple times).

### What steps can be taken to reproduce the issue? 
Hard to say. I am web scrapping some monthly data, processing them and storing to file. After each month is processed datas are uploaded to CKAN. When I download file from CKAN it is the same one as the one I uploaded there. Just the number of records is different than number of lines in file.

**Comment by smotornyuk** [comment]
Have you tried to re-upload file to datastore:
![image](https://user-images.githubusercontent.com/11091199/80384450-8f1d4000-88ad-11ea-8342-f9d07ab9b000.png)

**Comment by hrnciar** [comment]
@smotornyuk, I tried it now and it helped. Any ideas why it shows different number when I upload file using API?

**Comment by smotornyuk** [comment]
Datastore should be updated automatically when resources updated. If it doesn't happen, try to use [ckan.datapusher_formats](https://docs.ckan.org/en/2.8/maintaining/configuration.html?highlight=ckan.datapusher.formats#ckan-datapusher-formats). If this option doesn't work, the only thing I can suggest(because you are already using API) - make a call to [`datapusher_submit`](https://github.com/ckan/ckan/blob/master/ckanext/datapusher/plugin.py#L132-L136) action right after resource update.

**Comment by hrnciar** [comment]
Just for clarification, is it correct if I am uploading data using CKAN API like this? 

https://github.com/hrnciar/diplomka/blob/ee954e25a394164e9ed3cf256b042a1159eb8891/teplota/teplota.py#L271

Or should I be using datapusher methods?

**Comment by smotornyuk** [comment]
Yes, look correct. Try to add something like `ckan_post_request(config['url_api'], 'datapusher_submit', {'resource_id': resource_id}, headers)` right after all to `create/update` endpoint. But you either need to create new function, or update `ckan_post_request`, because it's has mandatory `filename`.

**Comment by hrnciar** [comment]
Seems like it didn't help. I haven't noticed any change with or without `datapusher_submit`.

I think the problem is with DataStore itself. For some reason, it pushes only some number of entries, like the rest of chunks was somehow lost. 
In this case  26203 entries out of 34943. 
~~~~
Upload Log
Fetching from: http://sc02.fi.muni.cz:80/dataset/7c9f7a37-a119-4d79-a203-5c42f30c36d8/resource/feb02e09-1632-4382-83dd-0aeae55a2558/download/teplota_2019.csv
Just now Details

Deleting "feb02e09-1632-4382-83dd-0aeae55a2558" from datastore.
Just now Details

Determined headers and types: [{'type': u'timestamp', 'id': u'2019-01-01T00:00:00'}, {'type': u'numeric', 'id': u'2.4'}, {'type': u'numeric', 'id': u'2.6'}]
Just now Details

Saving chunk 0
Just now Details

...

Saving chunk 104
Just now Details

Successfully pushed 26203 entries to "feb02e09-1632-4382-83dd-0aeae55a2558".
Just now Details

End of log
~~~~

**Comment by smotornyuk** [comment]
I've downloaded the mentioned file and there are ~35k actual data-rows, not just lines. But, when I'm uploading it to my local datastore(ckan 2.8.4, datapusher from master branch) everything works just as expected and I have no missing rows

**Comment by hrnciar** [comment]
So, I have finally tackled this issue. But I am not sure if it's the proper way how to handle it. I am storing ids of the updated datasets into a file. Then for all of the ids, I am calling `paster datapusher submit`.

~~~~
IFS=$'\t\n\r' read -d '' -r -a ids < ids.txt
for id in "${ids[@]}"
do
	echo $id
	yes | docker exec -i ckan /usr/local/bin/ckan-paster --plugin=ckan datapusher submit $id -c /etc/ckan/production.ini 
done

rm ids.txt
~~~~
